### PR TITLE
[DEV-1092] Add define auth challenge handler

### PR DESCRIPTION
**Pull request by datalek**
> [!IMPORTANT]  
> The lambda in not yet connected to the cognito trigger. This will be done later once the UI is ready

#### List of Changes

- Add `defineAuthChallengeHandler`
- Add terraform lambda resource

#### Motivation and Context

Thee login flow must have the MFA via email

#### How Has This Been Tested?

Unit test

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .infrastructure/55_cognito.tf [hunk 1]
```diff
@@ -64,6 +64,30 @@ module "cognito_post_confirmation_function" {
   }
 }
 
+module "cognito_define_auth_challenge_function" {
+  source = "terraform-aws-modules/lambda/aws"
+
+  function_name = "cognito_define_auth_challenge"
+  description   = "This Lambda function is invoked to create a challenge to present to the user."
```
Comment (datalek) [inline .infrastructure/55_cognito.tf]:
```suggestion
  description   = "This Lambda function is invoked to initiate the custom authentication flow."
```

File: apps/cognito-functions/src/define-auth-challenge-handler.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      const response = {
+        challengeName: 'PASSWORD_VERIFIER',
+        failAuthentication: false,
+        issueTokens: false,
+      };
+      return { ...event, response };
+    } else if (
+      session.length === 2 &&
+      session[1].challengeName === 'PASSWORD_VERIFIER' &&
+      session[1].challengeResult == true
```
Comment (kin0992) [inline apps/cognito-functions/src/define-auth-challenge-handler.ts:22]:
```suggestion
      session[1].challengeResult
```
this should work too

File: apps/cognito-functions/src/define-auth-challenge-handler.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      const response = {
+        challengeName: 'CUSTOM_CHALLENGE',
+        failAuthentication: false,
+        issueTokens: false,
+      };
+      return { ...event, response };
+    } else if (
+      session.length === 3 &&
+      session[2].challengeName === 'CUSTOM_CHALLENGE' &&
+      session[2].challengeResult === true
```
Comment (kin0992) [inline apps/cognito-functions/src/define-auth-challenge-handler.ts:35]:
```suggestion
      session[2].challengeResult
```
this should work too

**Review by kin0992 — APPROVED** [review]
🔏

File: apps/cognito-functions/src/define-auth-challenge-handler.ts (hunk 2 above)
Comment (datalek) [inline apps/cognito-functions/src/define-auth-challenge-handler.ts:22]:
```suggestion
      session[1].challengeResult === true
```

**Review by kin0992 — APPROVED** [review]

**Pull request merged**